bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/IContentAssistProcessorExtension2.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuła and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pauła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Extension interface of {@link IContentAssistProcessor} that forward information about
+ * autoActivation
+ *
+ * @since 3.24
+ */
+public interface IContentAssistProcessorExtension2 extends IContentAssistProcessor {
```

Comment:
```suggestion
public interface IContentAssistProcessorExtension2 extends IContentAssistProcessor2 {
```

it then should also replace/deprecate the methods there that currently take a viewer+offset.

Reply:
I don't think we should make the interface extend IContentAssistProcessor by default as it doesn't bring any value. We should document that IContentAssistProcess can be decorated with IContentAssistProcessor2 and annotation that IContentAssistProcessor2 can be implementing IContentAssistProcessor; but I see no immediate value in mandating the hierarchy between both.

Reply:
Parent is necessary if we want keep default methods. Without this Implementing both IContentAssistProcessor and IContentAssistProcessorExtensions2 will generate conflict

Reply:
I don't get it: why would we want to keep default methods here?
Using an IContentAssistProcessor instead of an IContentAssistProcessor2 would be relatively simple to do internally without exposing default methods, wouldn't it?

Reply:
OK, I see that we have the `IContentAssistant.getContentAssistProcessor(String contentType)` that we want to keep working with new implementations as well...